apps/src/templates/GameButtons.jsx

```diff
@@ -19,6 +19,47 @@ const styles = {
   }
 };
 
+// The following is a summary of the flow from clicking the finish button
+// to navigating to the next lesson/puzzle/congrats page.
+//
+// Note: this flow was traced in the context of a `studio` app and might be
+// slightly different for e.g. a `P5Lab` app.
+//
+// 1) The click handler for this button is added in `onMount` in studio.js
+// and calls `Studio.onPuzzleComplete`
+//
+// 2) `Studio.onPuzzleComplete` calls `StudioApp.report`, which then calls the
+// `onAttempt` function defined in `setupApp` in loadApp.js
+//
+// 3) `onAttempt` calls `reporting.sendReport` in reporting.js, which posts an
+// ajax request to the `milestone` endpoint in activities_controller.rb
+//
+// 4) `milestone` calls `milestone_response` in application_controller.rb,
+// which calls `script_level_solved_response` in script_level_helper.rb
+//
+// 5) `script_level_solved_response` calls `next_level_or_redirect_path_for_user`
+// in script_level.rb, which returns the path to navigate to and sets it as the
+// `redirect` property in the response
+//
+// 6) the `success` callback defined in `reporting.sendReport` calls the
+// `reportComplete` function defined in the same file, which sets
+// `lastServerResponse.nextRedirect` to the `redirect` path returned from the
+// `milestone` endpoint, then calls `Studio.onReportComplete`, which was passed
+// in as the `onComplete` callback in the call to `StudioApp.report` from
+// `Studio.onPuzzleComplete` way back in step 2
+//
+// 7) `Studio.onReportComplete` then calls `Studio.displayFeedback`, which calls
+// `StudioApp.displayFeedback`, which calls `FeedbackUtils.displayFeedback`
+// defined in feedback.js
+//
+// 8) `FeedbackUtils.displayFeedback` presents a `LegacyDialog` js component,
+// passing in the `onContinue` function defined in `setupApp` in loadApp.js
+// as the dialog's `onHidden` callback
+//
+// 9) The `onContinue` callback is called when the feedback dialog is dismissed,
+// which finally redirects to the `lastServerResponse.nextRedirect` path
+// set back in step 6
+//
```

Comment:
I might suggest putting this in the PR description rather than in the code. This describes how this flow works right now, but any part of this could change at any time (i.e. LegacyDialog has been deprecated and will likely be updated to Dialog at some point). Any time someone updates a part of this flow, they won't know to come here and update the comment. Since it's recording details about code that isn't nearby, the comment will become stale and may misdirect people. The PR description, however, is a point-in-time record of the code and is known to be stale as soon as the next PR is merged. Future developers won't be making similar assumptions about the PR description as they will about the code itself.

Reply:
@jmkulwik @clareconstantine is there a third option other than commenting in the code or putting it in the PR description? since this PR doesn't actually touch the process that i was documenting, it seems pretty unlikely that anyone will find the description of this PR when looking into how this stuff works in the future.

FWIW in my experience stale comments are better than no comments at all. usually it's pretty obvious when they are no longer accurate, but they might at least point you to the commit when the comment became stale (e.g. you see that we are no longer using `LegacyDialog` and find the commit where we updated to `Dialog`) so you can pick up the trail from there.

either way, this discussion makes me realize that this comment is unrelated to this PR so i will separate it out into its own PR where we can continue this discussion.